### PR TITLE
[SPARK-15025][SQL] fix duplicate of PATH key in datasource table options

### PR DESCRIPTION
**Pull request by xwu0226**
## What changes were proposed in this pull request?

The issue is that when the user provides the path option with uppercase "PATH" key, `options` contains `PATH` key and will get into the non-external case in the following code in `createDataSourceTables.scala`, where a new key "path" is created with a default path. 

```
val optionsWithPath =
      if (!options.contains("path")) {
        isExternal = false
        options + ("path" -> sessionState.catalog.defaultTablePath(tableIdent))
      } else {
        options
      }
```

So before creating hive table, serdeInfo.parameters will contain both "PATH" and "path" keys and different directories. and Hive table's dataLocation contains the value of "path". 

The fix in this PR is to convert `options` in the code above to `CaseInsensitiveMap` before checking for containing "path" key. 
## How was this patch tested?

A testcase is added

### CODE REVIEW

**Comment by xwu0226** [comment]
@yhuai @liancheng @liancheng @hvanhovell Can any of you help take a quick look at this change? Thank you very much!

**Comment by yhuai** [comment]
ok to test.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala [hunk 1]
```diff
@@ -1033,4 +1035,23 @@ class MetastoreDataSourcesSuite extends QueryTest with SQLTestUtils with TestHiv
...
+        sql(
+          s"""CREATE TABLE t USING PARQUET
+             |OPTIONS (PATH '$path')
+             |AS SELECT 1 AS a, 2 AS b, 3 AS c
+           """.stripMargin
+        )
+        sql("insert into t values (2, 3, 4)")
+        checkAnswer(table("t"), Seq(Row(1, 2, 3), Row(2, 3, 4)))
+        assert(sessionState.metadataHive.runSqlHive("select * from t order by a") ===
+          Seq("1\t2\t3", "2\t3\t4"))
```
Comment (yhuai) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala]: Can we avoid of calling `metadataHive.runSqlHive`? For this case, we want to check that there is a single entry in table properties with the key as `PATH` and the location is the one we want, right?

Comment (xwu0226) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala]: @yhuai Thanks for your input!. Yes, you are right. I can get the CatalogTable.storage.serdeProperties to check the 'PATH' key. I will modify the test case.

**Comment by xwu0226** [comment]
@yhuai I updated the testcase to check the key `path` from `CatalogTable.storage.serdeProperties`. In this case, we are supposed to see only `PATH` key-value. Let me know if you have any question. Thanks!

**Comment by yhuai** [comment]
test this please

**Comment by yhuai** [comment]
LGTM. Merging to master and branch 2.0.

**Pull request closed without merging**